pallets/settlement/src/lib.rs

```diff
@@ -1448,4 +1458,17 @@ impl<T: Trait> Module<T> {
                 .into()
         })
     }
+
+    // fn ensure_portfolios_and_affirmation_status(
+    //     instruction_id: u64,
+    //     portfolios: BTreeSet<PortfolioId>,
+    //     secondary_key: Option<&SecondaryKey<T::AccountId>>,
+    //     expected_affirmation_states: Vec<AffirmationStatus>
+    // ) -> DispatchResult {
+    //     for portfolio in &portfolios {
+    //         T::Portfolio::ensure_portfolio_custody_and_permission(*portfolio, did, secondary_key.as_ref())?;
+    //         let user_affirmation = Self::user_affirmations(portfolio, instruction_id);
+    //         ensure!(valid_affirmation_states.contains(user_affirmation), Error::<T>::UnexpectedAffirmationStatus);
+    //     }
+    // }
```

Comment:
What's happening here?

Reply:
Hmm. Seems like I started factoring out some common code and then got sick for a week :). Will either clean this up or complete this factoring out tomorrow.